### ISSUE TITLE
After sell timeout freqtrade sold at a loss

### ISSUE THREAD

**Issue by aristosv** [freqtrade/freqtrade#4681]
## Describe your environment

  * Operating system: Debian 10
  * Python Version: Python 3.7.3
  * CCXT version: ccxt==1.43.89
  * Freqtrade Version: freqtrade develop-8d01767a
  
## Your question

I've had a few cases, where freqtrade cancelled a sell order because of timeout, and then immediately sold with a loss. It's a bit weird, because if the sell order timed out, why would it try to sell again, but this time with a loss? Is there a way to prevent this from happening?

<img width="960" alt="1" src="https://user-images.githubusercontent.com/528302/113674297-b803a300-96c2-11eb-97f8-72584b2a01ea.png">

<img width="963" alt="2" src="https://user-images.githubusercontent.com/528302/113674333-c0f47480-96c2-11eb-988b-df10f4273903.png">

config.json
```
{
    "max_open_trades": -1,
    "stake_currency": "USDT",
    "stake_amount": 100,
    "fiat_display_currency": "USD",
    "dry_run": false,

    "unfilledtimeout": {
        "buy": 10,
        "sell": 30
    },

    "bid_strategy": {
        "ask_last_balance": 0.0,
        "use_order_book": false,
        "order_book_top": 1,
        "check_depth_of_market": {
            "enabled": false,
            "bids_to_ask_delta": 1
        }
    },

    "exchange": {
        "name": "binance",
        "key": "$key",
        "secret": "$secret",
        "ccxt_config": {"enableRateLimit": true},
        "ccxt_async_config": {
            "enableRateLimit": true,
            "rateLimit": 200
        },

        "pair_whitelist": [
        ],

        "pair_blacklist": [
            ".*UP/USDT",
            ".*DOWN/USDT",
            "BUSD/USDT",
            "EUR/USDT",
            "AUD/USDT",
            "BRL/USDT",
            "GBP/USDT",
            "TRY/USDT",
            "TUSD/USDT",
            "USDC/USDT",
            "PAX/USDT",
            "BIDR/USDT",
            "DAI/USDT",
            "IDRT/USDT",
            "RUB/USDT",
            "ZAR/USDT",
            "NGN/USDT",
            "UAH/USDT",
            "VAI/USDT",
            "BVND/USDT",
            "PAXG/USDT"
        ],

        "outdated_offset": 5,
        "markets_refresh_interval": 60
    },

    "pairlists": [
        {
            "method": "VolumePairList",
            "number_assets": 350,
            "sort_key": "quoteVolume",
            "refresh_period": 1800
        },
        {"method": "AgeFilter", "min_days_listed": 5},
        {"method": "PrecisionFilter"},
        {"method": "PriceFilter", "low_price_ratio": 0.01, "min_price": 0.00000010},
        {"method": "SpreadFilter", "max_spread_ratio": 0.005},
        {"method": "ShuffleFilter"},
        {
            "method": "RangeStabilityFilter",
            "lookback_days": 10,
            "min_rate_of_change": 0.03,
            "refresh_period": 1440
        }
    ],

    "protections": [
        {
            "method": "StoplossGuard",
            "lookback_period_candles": 60,
            "trade_limit": 4,
            "stop_duration_candles": 60,
            "only_per_pair": false
        },
        {
            "method": "CooldownPeriod",
            "stop_duration_candles": 1
        },
        {
            "method": "MaxDrawdown",
            "lookback_period_candles": 200,
            "trade_limit": 20,
            "stop_duration_candles": 10,
            "max_allowed_drawdown": 0.2
        },
        {
            "method": "LowProfitPairs",
            "lookback_period_candles": 360,
            "trade_limit": 2,
            "stop_duration_candles": 12,
            "required_profit": 0.02
        }
    ],

    "api_server": {
        "enabled": true,
        "listen_ip_address": "$ip",
        "listen_port": $port,
        "verbosity": "error",
        "enable_openapi": false,
        "jwt_secret_key": "$jwt",
        "CORS_origins": [],
        "username": "$user",
        "password": "$pass"
    },

    "telegram": {
        "enabled": true,
        "token": "$token",
        "chat_id": "$chatid",
        "keyboard": [
          ["/daily", "/profit", "/count"],
          ["/status table", "/stats", "/trades"]
      ]
    },

    "initial_state": "running",
}
```

strategy
```
from freqtrade.strategy.interface import IStrategy
from typing import Dict, List
from functools import reduce
from pandas import DataFrame
import talib.abstract as ta
import freqtrade.vendor.qtpylib.indicators as qtpylib
class freqtrade_USDT(IStrategy):

    stoploss = -0.3
    timeframe = '5m'
    trailing_stop = True
    trailing_stop_positive = 0.01
    trailing_stop_positive_offset = 0.02
    trailing_only_offset_is_reached = True

    def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        macd = ta.MACD(dataframe)
        dataframe['macd'] = macd['macd']
        dataframe['macdsignal'] = macd['macdsignal']
        dataframe['macdhist'] = macd['macdhist']
        dataframe['cci'] = ta.CCI(dataframe)
        dataframe['rsi'] = ta.RSI(dataframe)
        return dataframe

    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['rsi'] < 50) &
                qtpylib.crossed_above(dataframe['macd'], dataframe['macdsignal']) &
                (dataframe['cci'] <= -50.0)
            ),
            'buy'] = 1
        return dataframe

    def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        dataframe.loc[
            (
                (dataframe['rsi'] > 70) &
                qtpylib.crossed_below(dataframe['macd'], dataframe['macdsignal']) &
                (dataframe['cci'] >= 100.0)
            ),
            'sell'] = 1
        return dataframe
```

**Comment by xmatthias** [comment]
I would call this perfectly valid and intended behavior.

Based on your settings - price reached 2% profit - then dropped by 1% (which causes the initial trailing stoploss order).

As this order is not filled, the order is cancelled and replaced (-0.25% is below the trailing stoploss target of 1% (1ish) - so selling again is correct.

As you use limit orders (the default) without inversed sides (the default) - the sell limit order will happen at the best ask price - which (in case of a sudden drop) might not be reached again within the configured timeout (price does not recover up to your order).
The order is then cancelled and replaced (the stoploss is still active as the price is below the threshold) - and in this case, it's filled (the pricing is still the same, but the market might be recovering at this stage).

You have multiple ways to increase the possibility to get more fills 
- either by using market orders at the sell side (you'll pay the spread, and MUST inverse the pricing for this side to get correct results)
- alternatively, by reversing prices and staying with limit orders (this will increase fills, but not guarantee 100%)
- using stoploss on exchange - which will keep the stoploss order on the exchange -  so you'll usually not be "late".

in each of the above cases, you'll "victimize" some of the profits by paying the spread (except for stoploss on exchange - however the limit order is placed 1% below the stoploss by default - to (kindof) guarantee fills at the best price - which will in the end also result in paying the spread.

---

Apart from that, i'd doubt that you can be profitable long-term with this approach.

if you run into one initial stoploss of 30% - you'll need to win 30+ 1% trades to recover from this - and with the settings, most sells will probably be between 1 and 3% (crypto-markets often pull back by 1% ...).
This might work (or seem to work) in a bull market - but once market turns, and you run into 2-3 30% stoplosses, you'll have wiped out most of the previous profits almost certainly.

**Comment by aristosv** [comment]
> You have multiple ways to increase the possibility to get more fills
> 
> * either by using market orders at the sell side (you'll pay the spread, and MUST inverse the pricing for this side to get correct results)
> * alternatively, by reversing prices and staying with limit orders (this will increase fills, but not guarantee 100%)
> * using stoploss on exchange - which will keep the stoploss order on the exchange -  so you'll usually not be "late".

Thanks, I'll probably configure stoploss on exchange, it seems to be the better option.



> if you run into one initial stoploss of 30% - you'll need to win 30+ 1% trades to recover from this - and with the settings, most sells will probably be between 1 and 3% (crypto-markets often pull back by 1% ...).
> This might work (or seem to work) in a bull market - but once market turns, and you run into 2-3 30% stoplosses, you'll have wiped out most of the previous profits almost certainly.

Backtesting 365 days showed good results with this strategy, but I will also backtest with lower stoploss value (-10% and -20%). Thanks for the feedback.

**Comment by xmatthias** [comment]
> Backtesting 365 days showed good results with this strategy

make sure to be familiar with the [backtesting assumptions](https://www.freqtrade.io/en/latest/backtesting/#assumptions-made-by-backtesting) ... trailing stop  with rather tight values tends to overstate profits in backtesting - which will then (eventually) not reproduce in dry/live.
